client/time/time-ago.ts

```diff
@@ -0,0 +1,51 @@
+const NOW = Date.now()
```

Comment:
This doesn't actually work, this will be resolved to whatever time is "now" at the time this file is evaluated, not when the function is run.

I'd instead add another parameter to the `timeAgo` function that represents "now", and default it to `Date.now()` (defaults are executed when the function is called). That'd allow us to write tests for this easily but also make it simple to call if we don't care (note that it still might be good to call `Date.now()` once and pass that value for all further renders, just as a matter of efficiency

Reply:
Alternatively you could just change the function to take the difference in milliseconds directly, and leave it up to the caller to figure out how they want to calculate that (might be preferable if we e.g. have APIs that return a difference already)